app/src/main/java/com/pixlee/pixleeandroidsdk/ui/uioptions/ProductViewFragment.kt

```diff
@@ -0,0 +1,310 @@
+                }
+        )
+    }
+
+    fun openViwerActivity() {
+        lifecycleScope.launch {
+            withContext(Dispatchers.IO) {
+                intMockServer(getJson("pxl_product.json"))
+            }
+
+            val result = album.getFirstPage()
+            val info = PhotoWithVideoInfo(pxlPhoto = result.photos.first()!!,
+                    configuration = PXLPhotoView.Configuration().apply {
+                        // Customize image size, not a video
+                        pxlPhotoSize = PXLPhotoSize.ORIGINAL
+                        // Cystomize scale type
+                        imageScaleType = ImageScaleType.CENTER_CROP
+                        // Customize Main TextView
+                        mainTextViewStyle = TextViewStyle().apply {
+                            text = "Main Text"
+                            size = 30.px
+                            sizeUnit = TypedValue.COMPLEX_UNIT_PX
+                            typeface = null
+                        }
+                        // Customize Sub TextView
+                        subTextViewStyle = TextViewStyle().apply {
+                            text = "Sub Text"
+                            size = 18.px
+                            sizeUnit = TypedValue.COMPLEX_UNIT_PX
+                            typeface = null
+                        }
+                        // Customize Button
+                        buttonStyle = PXLPhotoView.ButtonStyle().apply {
+                            text = "Action Button"
+                            size = 20.px
+                            sizeUnit = TypedValue.COMPLEX_UNIT_PX
+                            typeface = null
+                            buttonIcon = com.pixlee.pixleesdk.R.drawable.baseline_play_arrow_white_24
+                            stroke = PXLPhotoView.Stroke().apply {
+                                width = 2.px.toInt()
+                                color = Color.WHITE
+                                radiusInPixel = 25.px
+                                padding = PXLPhotoView.Padding().apply {
+                                    left = 20.px.toInt()
+                                    centerRight = 40.px.toInt()
+                                    topBottom = 10.px.toInt()
+                                }
+                            }
+                        }
+                    },
+                    isLoopingVideo = true,
+                    soundMuted = true)
+            ViewerActivity.launch(context!!, info)
+        }
+    }
+
+    fun getJson(fileName: String): String {
+        return context?.assets?.open("json/$fileName")?.bufferedReader()?.use { it.readText() }
+                ?: ""
+    }
+
+    val mockWebServer: MockWebServer by lazy {
+        MockWebServer()
+    }
+
+    lateinit var basicDS: KtxBasicDataSource
+    lateinit var analyticsDS: KtxAnalyticsDataSource
+    fun setupMockedWebServer() {
```

Comment:
Is this code for tests or demos? Not sure if it is worth the hassle, but if it is you might prefer to put this in an extension function in a different file, and only conditionally compile that file using `sourceSets.main.java.exclude "filename.kt"`.

Reply:
- all codes in /app directory are for the demo app. 
- This is for the demo app but it mocks the server and emmit the response body from a JSON file stored in the project. It's worth moving the unnecessary codes to another wrapper class. I'll move the mock server things to a new class. Thanks @jharrilim